contrib/epee/src/readline_buffer.cpp

```diff
@@ -196,15 +200,50 @@ static int startup_hook()
   return 0;
 }
 
+static char* completion_matches(const char* text, int state)
+{
+  static size_t list_index;
+  static size_t len;
+
+  if(state == 0)
+  {
+    list_index = 0;
+    len = strlen(text);
+  }
+
+  const std::vector<std::string>& completions = rdln::readline_buffer::get_completions();
+  for(; list_index<completions.size(); )
+  {
+    const std::string& cmd = completions[list_index++];
+    if(cmd.compare(0, len, text) == 0)
```

Comment:
You _have_ to use the statics here as the function is called multiple times and you need to keep track of last call(s). Again, horrible readline interface but that's what we have to work with!